system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java

```diff
@@ -25,6 +28,23 @@ public void testNodeDiscoveryRest() throws InterruptedException {
 
         ElasticsearchNodesResponse nodesResponse = new ElasticsearchNodesResponse(tasks, NODE_COUNT);
         assertTrue("Elasticsearch nodes did not discover each other within 5 minutes", nodesResponse.isDiscoverySuccessful());
-    }
+
```

Comment:
I numbered your comments.

1. We don't test from inside the container. We only test the volume inside mesos-local.

2. That is tested here. The host machine is mesos-local

3. The volume is removed in https://github.com/ContainerSolutions/mini-mesos/blob/master/src/main/java/org/apache/mesos/mini/container/AbstractContainer.java#L105 

4. This is not supported yet. We would have to update the ClusterMonitor and the Scheduler.resourceOffers method to support this. The alternative is to use persistent volumes but those are blocked on authentication.